io/net/net_tools.py

```diff
@@ -0,0 +1,359 @@
+        if "default via" not in interface_out:
+            self.skip("No active interface with deafult gateway configured")
+        install_dependencies()
+        search_obj = re.search(
+            "^default via\s+(\S+)\s+dev\s+(\w+)", interface_out)
+        self.default_router = search_obj.group(1)
+
+    @avocado.fail_on(process.CmdError)
+    def test_arp(self):
+        """
+        Test to resolve Mac addr of default gateway router using arp
+        """
+        process.system("ping -c 2 -w 5 %s" % self.default_router)
+        output = process.run(cmd="arp -n", ignore_status=True,
+                             env={"LANG": "C"})
+        if output.exit_status:
+            self.fail("Arp reported non zero exit status")
+        if self.default_router not in output.stdout:
+            self.fail("unexpected response from arp")
+
+    def tearDown(self):
+        pass
+
+
+class NetworkUtilities(Test):
+    """
+    traceroute,traceroute6: run for localhost. It should report 1 hop.
+    route: run it with -n. Check the return code.
+    ipmaddr: run it and check the return code.
+    """
+
+    def setUp(self):
+        install_dependencies()
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+
+    @avocado.fail_on(process.CmdError)
+    def test_traceroute(self):
+        """
+        Verify traceroute,traceroute6 functionality.
+        """
+        ret = process.run(cmd="traceroute localhost",
+                          env={"LANG": "C"})
+        no_of_hops = re.search(
+            "(\d+)\s+\S+\s*\(127.0.0.1\)",
+            ret.stdout).group(1)
+        # Only one hop is required to get to localhost.
+        if str(no_of_hops) != '1':
+            self.fail("traceroute did not show 1 hop for localhost")
+
+        if self.ipv6:
+            detected_distro = distro.detect()
+            if detected_distro.name in ("SuSE", "Ubuntu"):
+                ret = process.run(cmd="traceroute6 ipv6-localhost",
+                                  env={"LANG": "C"})
+            else:
+                ret = process.run(cmd="traceroute6 localhost6",
+                                  env={"LANG": "C"})
+            no_of_hops = re.search(
+                "(\d+)\s+\S+\s*\(::1\)",
+                ret.stdout).group(1)
+            if str(no_of_hops) != '1':
+                self.fail(
+                    "traceroute6 did not show 1 hop for " +
+                    "localhost6/ipv6-localhost")
+
+    @avocado.fail_on(process.CmdError)
+    def test_netstat(self):
+        """
+        Verify the functionality of netstat
+        """
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('netstat_opt', default="s")
+        for option in options_to_verify:
+            ret = process.run("netstat -%s" % option, verbose=False,
+                              ignore_status=True)
+            if ret.exit_status:
+                self.fail(
```

Comment:
this is quite ugly, you can break the statements:

```
    self.fail("Foo...,"
              "bar %s,"
              "baz %s"
              % (1, 2))
```
which then becomes `Foo...,bar 1,baz2` and looks much nicer